### ISSUE TITLE
Create testcases for parts of NN training

### ISSUE THREAD

**Issue by acramer** [acramer/pyml#9]
- [x] Error Calc
- [ ] Gradient Calculation
- [x] Backprop
- [ ] Optimizer
- [ ] Optimizer multiple epoch
